### PR TITLE
Issue 5933: Checkpoint should not be empty

### LINKED ISSUE #5933: Checkpoint might be empty
**Describe the bug**
A `Stream names that the reader group can read from cannot be empty` error happened during Flink longevity run. After checking the code, it might be the case that `initiateCheckpoint` returns a `checkpoint` that does not contain any stream information. Currently there is no guarantee that the `checkpoint` cannot be empty, it's better to add an `assert` clause to fail fast before it returns to the caller.

**To Reproduce**
Flink longevity run. It looks like a rare case, no clear steps to reproduce.

### PR DESCRIPTION
**Pull request by Tristan1900**
**Change log description**  
Add an `checkArgument` clause to fail fast if the `checkpoint` is going to be empty. 
The `checkpoint` is constructed by using the content of `segmentPositions` map, we can just check if `segmentPositions` is empty.

**Purpose of the change**  
Fix #5933

### CODE REVIEW

File: client/src/main/java/io/pravega/client/stream/impl/CheckpointImpl.java [hunk 1]
```diff
@@ -45,6 +46,7 @@
     private final Map<Stream, StreamCut> positions;
     
     CheckpointImpl(String name, Map<Segment, Long> segmentPositions) {
+        Preconditions.checkArgument(!segmentPositions.isEmpty(), "segment positions should not be empty");
```
Comment (shrids) [inline client/src/main/java/io/pravega/client/stream/impl/CheckpointImpl.java:49]: could we add the checkpoint name here for easier debugging.

**Review by shrids — COMMENTED** [review]
LGTM...

**Review by shrids — APPROVED** [review]

**Comment by shrids** [comment]
@Tristan1900 the codecov report is stuck for the changeset :sweat_smile: . Could you try doing a force push into your branch to retrigger the build?

**Comment by RaulGracia** [comment]
I'm observing this `DataCorruptionException` exception consistently in this branch:
```
io.pravega.segmentstore.server.host.NonAppendExtendedS3IntegrationTest > testEndToEndWithChunkedStorage FAILED
    java.util.concurrent.CompletionException: io.pravega.segmentstore.server.DataCorruptionException: Unable to add operation 'StreamSegmentAppendOperation: SequenceNumber = 1680, SegmentId = 626, Offset = 0, Length = 56, Attributes = 11' to ReadIndex.
        at io.pravega.common.concurrent.Futures.lambda$join$2(Futures.java:209)
        at io.pravega.common.Exceptions.handleInterruptedCall(Exceptions.java:144)
        at io.pravega.common.concurrent.Futures.join(Futures.java:205)
        at io.pravega.segmentstore.server.mocks.LocalSegmentContainerManager.initialize(LocalSegmentContainerManager.java:119)
        at io.pravega.segmentstore.server.store.ServiceBuilder.initialize(ServiceBuilder.java:247)
        at io.pravega.segmentstore.server.store.StreamSegmentStoreTestBase.createBuilder(StreamSegmentStoreTestBase.java:513)
        at io.pravega.segmentstore.server.store.StreamSegmentStoreTestBase.endToEndProcess(StreamSegmentStoreTestBase.java:339)
        at io.pravega.segmentstore.server.store.StreamSegmentStoreTestBase.testEndToEndWithChunkedStorage(StreamSegmentStoreTestBase.java:291)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.lang.Thread.run(Thread.java:834)

        Caused by:
        io.pravega.segmentstore.server.DataCorruptionException: Unable to add operation 'StreamSegmentAppendOperation: SequenceNumber = 1680, SegmentId = 626, Offset = 0, Length = 56, Attributes = 11' to ReadIndex.
            at io.pravega.segmentstore.server.logs.MemoryStateUpdater.addToReadIndex(MemoryStateUpdater.java:275)
            at io.pravega.segmentstore.server.logs.MemoryStateUpdater.process(MemoryStateUpdater.java:195)
            at io.pravega.segmentstore.server.logs.RecoveryProcessor.recoverOperation(RecoveryProcessor.java:196)
            at io.pravega.segmentstore.server.logs.RecoveryProcessor.recoverAllOperations(RecoveryProcessor.java:165)
            at io.pravega.segmentstore.server.logs.RecoveryProcessor.performRecovery(RecoveryProcessor.java:101)
            at io.pravega.segmentstore.server.logs.DurableLog.performRecovery(DurableLog.java:221)
            at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
            at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
            at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
            at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
            ... 1 more

            Caused by:
            java.lang.IllegalArgumentException: New generation must be at least the value of the previous one.
                at com.google.common.base.Preconditions.checkArgument(Preconditions.java:142)
                at io.pravega.segmentstore.server.reading.ReadIndexSummary.setCurrentGeneration(ReadIndexSummary.java:55)
                at io.pravega.segmentstore.server.reading.StreamSegmentReadIndex.updateGenerations(StreamSegmentReadIndex.java:234)
                at io.pravega.segmentstore.server.CacheManager.register(CacheManager.java:187)
                at io.pravega.segmentstore.server.reading.ContainerReadIndex.getOrCreateIndex(ContainerReadIndex.java:406)
                at io.pravega.segmentstore.server.reading.ContainerReadIndex.append(ContainerReadIndex.java:130)
                at io.pravega.segmentstore.server.logs.MemoryStateUpdater.addToReadIndex(MemoryStateUpdater.java:253)
                ... 12 more
```
Maybe it is not related, but is the first time I see this in a build. We need to look into it before merging.

**Pull request merged**